source/extensions/filters/common/expr/evaluator.h

```diff
@@ -44,6 +46,7 @@ class StreamActivation : public google::api::expr::runtime::BaseActivation {
 
 protected:
   void resetActivation() const;
+  mutable const ::Envoy::LocalInfo::LocalInfo* local_info_{nullptr};
```

Comment:
I think this can be const ref and added to the default constructor since there's always only one local info.

Reply:
not sure I understand correctly, but PTAL.

Reply:
I mean, local info is always available AFAICT, so you could make it explicit:

```
class StreamActivation {
 StreamActivation(const LocalInfo& local_info) local_info_(local_info) {}
 const LocalInfo& local_info_;
};
```

Reply:
I recall there's too much work to pass `LocalInfo` in somewhere, but anyway I will give a try.

Reply:
It seems that changing the default constructor causes more problems than expected.

Reply:
OK, I see that you have to change the factory signatures in many places . Sorry about the churn then, let's roll back to using `nullptr` like before and then patch it up later. I'm actually not sure why `LocalInfo` is not a singleton to begin with.

Reply:
yeah, because one thing that should be simple, many seemingly unrelated things need to be changed.